src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -768,8 +770,9 @@ def get_result_set(
     def get_result_and_metadata(
         self, plan: SnowflakePlan, **kwargs
     ) -> Tuple[List[Row], List[Attribute]]:
-        result_set, result_meta = self.get_result_set(plan, **kwargs)
-        result = result_set_to_rows(result_set["data"])
+        show_limit = kwargs.pop("show_limit", None)
```

Comment:
I'm slightly inclined towards making the parameter part of the function signature and let kwargs be same as what users passed in

```
def get_result_and_metadata(
        self, plan: SnowflakePlan, limit: Optional[int] = None, **kwargs
    ) -> Tuple[List[Row], List[Attribute]]:
```